### PR TITLE
fix flake test with GetWorksByBindingID func

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

After merging #5742, there is a flake test in the master branch, this is because of the change of #5779, as commented in https://github.com/karmada-io/karmada/pull/5742#issuecomment-2566086990

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/helper/work_test.go [hunk 1]
```diff
@@ -309,54 +309,62 @@ func TestGetWorksByBindingID(t *testing.T) {
 	assert.NoError(t, workv1alpha1.Install(scheme))
 
 	bindingID := "test-binding-id"
-	work1 := &workv1alpha1.Work{
+	workWitRBID := &workv1alpha1.Work{
```
Comment (RainbowMango) [inline pkg/util/helper/work_test.go:312]:
```suggestion
	workWithRBID := &workv1alpha1.Work{
```

Comment (XiShanYongYe-Chang) [reply pkg/util/helper/work_test.go]: thx, updated.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**